mobile/cmp/navigator/NavigatorModel.ts

```diff
@@ -52,48 +57,58 @@ export class NavigatorModel extends HoistModel {
     stack: PageModel[] = [];
 
     pages: PageConfig[] = [];
-
     track: boolean;
-    swipeToGoBack: boolean;
     pullDownToRefresh: boolean;
+    transitionMs: number;
     renderMode: RenderMode;
     refreshMode: RefreshMode;
 
-    private _navigator = null;
+    private _swiper: Swiper;
     private _callback: () => void;
-    private _prevKeyStack: string[];
+    private _touchStartX: number;
 
     get activePageId(): string {
         return this.activePage?.id;
     }
 
     get activePage(): PageModel {
-        const {stack} = this;
-        return stack[stack.length - 1];
+        return this.stack[this.activePageIdx];
+    }
+
+    get activePageIdx(): number {
+        return this._swiper?.activeIndex ?? 0;
+    }
+
+    get allowSlideNext(): boolean {
+        return this.activePageIdx < this.stack.length - 1;
+    }
+
+    get allowSlidePrev(): boolean {
+        return this.activePageIdx > 0;
     }
 
     constructor({
         pages,
         track = false,
-        swipeToGoBack = true,
         pullDownToRefresh = true,
+        transitionMs = 500,
         renderMode = 'lazy',
         refreshMode = 'onShowLazy'
     }: NavigatorConfig) {
         super();
         makeObservable(this);
-        warnIf(
+        throwIf(
```

Comment:
this could probably be removed now that we have this throwIf on the page itself.